samples/matter/common/src/diagnostic/diagnostic_logs_provider.cpp

```diff
@@ -123,11 +138,15 @@ CHIP_ERROR DiagnosticLogProvider::EndLogCollection(LogSessionHandle sessionHandl
 	/* Do the specific action for the removing intent */
 	switch (mIntentMap[sessionHandle].mIntent) {
 	case IntentEnum::kEndUserSupport:
-#ifndef CONFIG_NCS_SAMPLE_MATTER_DIAGNOSTIC_LOGS_TEST
-		/* TODO: add support for End User Intent */
+#ifdef CONFIG_NCS_SAMPLE_MATTER_DIAGNOSTIC_LOGS_END_USER_LOGS
+		if (mIntentMap[sessionHandle].mIntentImpl) {
```

Comment:
Theoretically we should check if the map `Contains()` the item we want to retrieve with [] operator.

Reply:
It is done few lines above:

VerifyOrReturnError(mIntentMap.Contains(sessionHandle), CHIP_ERROR_INTERNAL);